client/src/components/App.js

```diff
@@ -9,23 +9,31 @@ class App extends Component {
     super(props);
     this.state = {
       projects: [],
-      show_form: false,
+      modal: {type: '', active_project: {}},
     };
     this.showProjects = this.showProjects.bind(this);
     this.showForm = this.showForm.bind(this);
     this.hideForm = this.hideForm.bind(this);
+    this.submitNewProject = this.submitNewProject.bind(this);
+    this.submitEditedProject = this.submitEditedProject.bind(this);
+    this.inspectProject = this.inspectProject.bind(this);
+    this.parseProjectForm = this.parseProjectForm.bind(this);
   }
 
   UNSAFE_componentWillMount() {
     this.showProjects();
   }
 
+  inspectProject(active_project) {
+    this.setState({modal: {type: 'details', active_project}});
+  }
+
   showForm() {
-    this.setState({show_form: true});
+    this.setState({modal: {type: 'new', active_project: {}}});
```

Comment:
If the function above is `inspectProject`, should this be `createProject`, since they both show a form?